### ISSUE TITLE
Is dataLayer suposed to be saved from previous visits?

### ISSUE THREAD

**Issue by gety9** [yireo/Yireo_GoogleTagManager#64]
On the first visit of the home page i have the following dataLayer:
`[({"customerLoggedIn":0,"customerId":0,"customerGroupId":"1","customerGroupCode":"GENERAL","pageType":"cms\/index\/index"})];`

When placing the test order, and than closing the browser and revisting the home page it shows the following dataLayer:
`[({"customerLoggedIn":0,"customerId":0,"customerGroupId":"1","customerGroupCode":"GENERAL","transactionEntity":"ORDER","transactionId":"193552702","transactionDate":"2017-09-06 21:50:23","transactionAffiliation":"Main Website","transactionTotal":70.2,"transactionSubtotal":78,"transactionTax":-7.8,"transactionShipping":0,"transactionPayment":"Check \/ Money order","transactionCurrency":"USD","transactionPromoCode":"OFFNA10","transactionProducts":[{"id":"60","sku":"handbag24","name":"Black Handbag by Marci","price":0,"priceexcludingtax":"0.00","tax":"0.00","taxrate":0,"type":"bundle","category":"All Tours","quantity":"1.0000"}],"pageType":"cms\/index\/index"})];`

Basically it kind of saves dataLayer values from previous visit(order). Is it how it is supposed to work?

All cache is disabled.

**Comment by jissereitsma** [comment]
The GTM data simply follows whatever the Magento system supplies based on cookies. And the cookies are not managed by our GTM extension but by other Magento modules. Can you confirm that the cookies are also still alive? If so, that's the answer: As long as Magento keeps remember the previous order ID, that is most likely done by intention of the Magento system, and our GTM extension should simply follow. As a remark: In any Magento shop that I manage, the last order is wiped out by the payment gateway used to finalize the order, and then by the success page. So either way, the problem seem you to face is related to customization, not related to generic Magento behaviour. It might be best to have a developer look into this.

**Comment by gety9** [comment]
Thank you for your reply. If it's based on the cookies than everything is working correctly.

I just thought there is some kind of problem with my set-up.

Thank you for fast reply and great extension!
